From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:17:19 -0700
Subject: [PATCH] ==== drgn and python based sysfs selftest ====

selftests/damon: add python and drgn based DAMON sysfs functionality tests

DAMON sysfs interface is the bridge between the user space and the
kernel space for DAMON parameters.  There is no good and simple test to
see if the parameters are set as expected.  Existing DAMON selftests
therefore test end-to-end features.  For example, damos_quota_goal.py
runs a DAMOS scheme with quota goal set against a test program running
an artificial access pattern, and see if the result is as expected.
Such tests cover only a few part of DAMON.  Adding more tests is also
complicated.  Finally, the reliability of the test itself on different
systems is bad.

'drgn' is a tool that can extract kernel internal data structures like
DAMON parameters.  Add a test that passes specific DAMON parameters via
DAMON sysfs reusing _damon_sysfs.py, extract resulting DAMON parameters
via 'drgn', and compare those.  Note that this test is not adding
exhaustive tests of all DAMON parameters and input combinations but very
basic things.  Advancing the test infrastructure and adding more tests
are future works.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 .../testing/selftests/damon/hkml_cv_bogus/hkml_cv_bogus__tt9_i4c  | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 tools/testing/selftests/damon/hkml_cv_bogus/hkml_cv_bogus__tt9_i4c

diff --git a/tools/testing/selftests/damon/hkml_cv_bogus/hkml_cv_bogus__tt9_i4c b/tools/testing/selftests/damon/hkml_cv_bogus/hkml_cv_bogus__tt9_i4c
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

